rogue/src/components/status_effects.rs

```diff
@@ -112,25 +133,39 @@ pub fn tick_status_with_immunity<Status, StatusImmune>(
     }
 }
 
+pub fn remove_status<Status: Component + StatusEffect>(
+    store: &mut WriteStorage<Status>,
+    entity: Entity,
+) {
+    let status = store.get_mut(entity);
+    if let Some(_) = status {
+        store.remove(entity);
+    }
+}
+
+
 //------------------------------------------------------------------
 // Status Effect Components.
 //------------------------------------------------------------------
 pub trait StatusEffect {
-    fn new(turns: i32) -> Self;
+    fn new(turns: i32, render_glyph: bool) -> Self;
     fn remaining_turns(&self) -> i32;
     fn set_remaining_turns(&mut self, turns: i32);
+    fn do_render(&self) -> bool;
 }
 
 // Component indicating the entity is frozen. A frozen entity cannot take any
 // action until the status has expired, or is removed.
 #[derive(Component, ConvertSaveload, Clone)]
 pub struct StatusIsFrozen {
     pub remaining_turns: i32,
+    pub render_glyph: bool
 }
 impl StatusEffect for StatusIsFrozen {
```

Comment:
As far as I can tell, all of the `StatusEffect` implementations are the exact same. Is that something you expect might change over time, or is this somewhere where either making `StatusEffect` a derive-able trait, or using a single implementation plus something like [newtypes](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) could help reduce the duplication?

Reply:
Ahh. Yah, I had the feeling that there was a better way to do this, and you put me towards it here. I'll read the book chapter on traits and derivation.